src/daemon/client.c

```diff
@@ -293,7 +293,18 @@ client_handle_set_port(struct lldpd *cfg, enum hmsg_type *type,
 	TAILQ_FOREACH(hardware, &cfg->g_hardware, h_entries)
 	    if (!strcmp(hardware->h_ifname, set->ifname)) {
 		    struct lldpd_port *port = &hardware->h_lport;
-		    (void)port;
+		    if (set->p_id && strlen(set->p_id) > 0) {
+			    log_debug("rpc", "requested change to Port ID");
+			    if (port->p_id) {
+				    free(port->p_id);
+			    }
+			    port->p_id = strdup(set->p_id);
+			    port->p_id_len = strlen(set->p_id);
+			    port->p_id_subtype = LLDP_PORTID_SUBTYPE_LOCAL;
+		    } else {
+			    /* This is the default value */
+			    port->p_id_subtype = LLDP_PORTID_SUBTYPE_LLADDR;
```

Comment:
Moreover, setting the type back to LLADDR is not enough since the value is incorrect. Since currently, the code cannot be hit, just don't do anything if the provided local ID is empty.